graylog2-server/src/main/java/org/graylog2/outputs/ElasticSearchOutput.java

```diff
@@ -94,18 +96,26 @@ public void write(List<Message> messageList) throws Exception {
 
     public void writeMessageEntries(List<Map.Entry<IndexSet, Message>> messageList) throws Exception {
         if (LOG.isTraceEnabled()) {
-            final List<String> sortedIds = Ordering.natural().sortedCopy(messageList.stream()
-                    .map(entry -> entry.getValue().getId())
-                    .collect(Collectors.toList()));
-            LOG.trace("Writing message ids to [{}]: <{}>", NAME, Joiner.on(", ").join(sortedIds));
+            final String sortedIds = messageList.stream()
+                    .map(Map.Entry::getValue)
+                    .map(Message::getId)
+                    .sorted(Comparator.naturalOrder())
+                    .collect(Collectors.joining(", "));
+            LOG.trace("Writing message ids to [{}]: <{}>", NAME, sortedIds);
         }
 
         writes.mark(messageList.size());
+        final List<String> failedMessageIds;
         try (final Timer.Context ignored = processTime.time()) {
-            messages.bulkIndex(messageList);
+            failedMessageIds = messages.bulkIndex(messageList);
         }
+        failures.mark(failedMessageIds.size());
+
         for (final Map.Entry<IndexSet, Message> entry : messageList) {
-            journal.markJournalOffsetCommitted(entry.getValue().getJournalOffset());
+            final Message message = entry.getValue();
+            if (!failedMessageIds.contains(message.getId())) {
+                journal.markJournalOffsetCommitted(message.getJournalOffset());
```

Comment:
This is not working the way it is supposed to. It just marks all messages up to and including the successfull message with the highest journal offset, including failed messages with lower journal offsets.